### ISSUE TITLE
Can Not Restore with -type=immediate --target-action=promote

### ISSUE THREAD

**Issue by bradnicholson** [pgbackrest/pgbackrest#496]
pgBackRest version: 2.00
PostgreSQL version: 9.6.7

when trying to perform a restore with the options `--type=immediate` and  `--target-action=promote`, it fails validation with 

`ERROR: [031]: option 'target-action' not valid without option 'type' in ('name', 'time', 'xid')`

I would think that we should be able to combine these two options.

The command line to trigger it is:

```
/usr/bin/pgbackrest --config=/conf/pgbackrest.conf   --stanza=test --type=immediate --target-action=promote  restore
```

**Comment by dwsteele** [comment]
Yeah, it looks like this got missed when `recovery_target = 'immediate'` was added in 9.4.

We'll fix this for 2.01, but in the meantime you can manually add it to `recovery.conf` after the restore.  How painful is this for you?

**Comment by bradnicholson** [comment]
Not painful at all.  I just discovered it while doing some testing.

**Comment by dwsteele** [comment]
Well, that's good.  The fix has been committed at 5e0ce3e996db90c44.

The next planned release is early April.  It may happen sooner if there is a serious bug reported, but this one has a fairly easy workaround so it doesn't seem to require a release.

Thanks for the report!
